src/elm/Data/DataSource.elm

```diff
@@ -0,0 +1,102 @@
+module Data.DataSource
+    exposing
+        ( DataSourceId(..)
+        , DataSource
+        , decode
+        , decodeList
+        , encodeId
+        )
+
+import Json.Decode as Decode exposing (Decoder)
+import Json.Encode
+
+
+type DataSourceId
+    = DataSourceId Int
+
+
+type Description
+    = NoDescription
+    | Description String
+
+
+type alias DataSource =
+    { id : DataSourceId
+    , title : String
+    , description : Description
+    }
+
+
+decodeList : Decoder (List DataSource)
+decodeList =
+    Decode.list decode
+
+
+decode : Decoder DataSource
+decode =
```

Comment:
Thoughts on naming this `decoder` ? I think using a noun rather than a verb better describes what this function returns. Decoders don't actually _do_ anything. The actual decoding gets done by functions like `Decoder.decodeString` or `Decoder.decodeValue`.